src/axolotl/utils/data/lock.py

```diff
@@ -46,18 +46,20 @@ def load(self, load_fn: Callable[[], Any]) -> Any:
     def _increment_counter(self):
         """Safely increment the process counter."""
         if self.counter_path.exists():
-            count = int(self.counter_path.read_text().strip())
+            counter_content = self.counter_path.read_text().strip()
+            count = int(counter_content) if counter_content else 0
         else:
             count = 0
```

Comment:
Should we make a simple `_get_counter` which handles this check if exists? Then we can re-use it in the `cleanup` as well.